modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientHandlerModule.java

```diff
@@ -198,17 +200,24 @@ protected void initChannel(Channel ch) {
                             ch.pipeline().addLast(new IdleChannelHandler());
                         }
 
-                        ch.pipeline().addLast(
-                                new ClientMessageDecoder(),
-                                new ClientInboundMessageHandler(
-                                        igniteTables,
-                                        igniteTransactions,
-                                        queryProcessor,
-                                        configuration,
-                                        igniteCompute,
-                                        clusterService,
-                                        sql,
-                                        clusterId));
+                        if (configuration.ssl().enabled()) {
+                            SslContext sslContext =  SslContextProvider.createServerSslContext(configuration.ssl());
+
+                            ch.pipeline().addFirst("ssl", sslContext.newHandler(ch.alloc()));
+                        }
+
+                        ch.pipeline()
+                                .addLast(
```

Comment:
This formatting is little bit weird, I think that the previous version was better